### PR TITLE
New Podcast Episode: Oxide with Steve Klabnik

### PR DESCRIPTION
**Pull request by mre**

### CODE REVIEW

**Comment by mre** [comment]
@prscheduler 14/11/2024T16:00

File: content/podcast/s03e03-oxide/index.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+- [cargo-xtask](https://github.com/matklad/cargo-xtask)
+- [Hubris Build Documentation](https://github.com/oxidecomputer/hubris#build)
+- [Buck Build System](https://buck.build/)
+- [Buildomat](https://github.com/oxidecomputer/buildomat)
+- [Omicron](https://github.com/oxidecomputer/omicron)
+- [illumos](https://illumos.org/)
+- [bhyve](https://bhyve.org/)
+- [About Self-hosted Runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners)
+- [Async Drop Initiative](https://rust-lang.github.io/async-fundamentals-initiative/roadmap/async_drop.html)
+- [Rust Playground Example for async Rust syntax](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=async+fn+f()+%7B%0A++++await+f()%0A%7D)
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Rust Playground Example for helpful error when using prefix `await` operator](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=async+fn+f()+%7B%0A++++await+f()%0A%7D)
```

File: content/podcast/s03e03-oxide/index.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+- [Oxide RFD 1](https://rfd.shared.oxide.computer/rfd/0001)
+- [Rust RFCs](https://rust-lang.github.io/rfcs/)
+- [IETF RFCs](https://en.wikipedia.org/wiki/Request_for_Comments)
+- [Zig](https://ziglang.org/)
+- [TigerBeetle](https://tigerbeetle.com/)
+- [Bun](https://bun.sh/)
+- [CockroachDB](https://www.cockroachlabs.com/)
+- [Oxide and Friends: Wither CockroachDB?](https://oxide.computer/podcasts/oxide-and-friends/2052742)
+- [Mozilla Public License](https://choosealicense.com/licenses/mpl-2.0/)
+- [Asahi Linux Rust Drivers](https://asahilinux.org/)
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Asahi Linux uses Rust to write drivers](https://asahilinux.org/)
```

**Review by M3t0r — CHANGES_REQUESTED** [review]

**Comment by mre** [comment]
@prscheduler 14/11/2024T18:00

File: content/podcast/s03e03-oxide/index.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- [bhyve](https://bhyve.org/) - Hypervisor
+- [About Self-hosted Runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) - GitHub Actions documentation
+- [Async Drop Initiative](https://rust-lang.github.io/async-fundamentals-initiative/roadmap/async_drop.html) - Rust async development
+- [Rust Playground Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=async+fn+f()+%7B%0A++++await+f()%0A%7D) - Demonstrating helpful error when using prefix `await` operator
+- [Rust Book - Modules](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html) - Rust module system
+- [OpenAPI Specification](https://swagger.io/specification/) - API documentation standard
+- [Dropshot](https://github.com/oxidecomputer/dropshot/) - Oxide's HTTP framework
+- [Axum](https://github.com/tokio-rs/axum) - Web framework for Rust
+- [Oxide Console](https://github.com/oxidecomputer/console) - Oxide's web interface
+- [Oxide Console Preview](https://console-preview.oxide.computer/projects) - Demo of Oxide Console
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Oxide Console Preview](https://console-preview.oxide.computer/projects) - Demo of Oxide Console using a mocked backend
```

File: content/podcast/s03e03-oxide/index.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+- [Buildomat](https://github.com/oxidecomputer/buildomat) - Oxide's build system
+- [Omicron](https://github.com/oxidecomputer/omicron) - Oxide's manufacturing test framework
+- [illumos](https://illumos.org/) - Unix operating system
+- [bhyve](https://bhyve.org/) - Hypervisor
+- [About Self-hosted Runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) - GitHub Actions documentation
+- [Async Drop Initiative](https://rust-lang.github.io/async-fundamentals-initiative/roadmap/async_drop.html) - Rust async development
+- [Rust Playground Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=async+fn+f()+%7B%0A++++await+f()%0A%7D) - Demonstrating helpful error when using prefix `await` operator
+- [Rust Book - Modules](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html) - Rust module system
+- [OpenAPI Specification](https://swagger.io/specification/) - API documentation standard
+- [Dropshot](https://github.com/oxidecomputer/dropshot/) - Oxide's HTTP framework
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Dropshot](https://github.com/oxidecomputer/dropshot/) - Oxide's OpenAPI server framework
```

File: content/podcast/s03e03-oxide/index.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+- [Axum](https://github.com/tokio-rs/axum) - Web framework for Rust
+- [Oxide Console](https://github.com/oxidecomputer/console) - Oxide's web interface
+- [Oxide Console Preview](https://console-preview.oxide.computer/projects) - Demo of Oxide Console
+- [Oxide RFD 1](https://rfd.shared.oxide.computer/rfd/0001) - Request for Discussion process
+- [Rust RFCs](https://rust-lang.github.io/rfcs/) - Rust's design process
+- [IETF RFCs](https://en.wikipedia.org/wiki/Request_for_Comments) - Internet standards process
+- [Zig](https://ziglang.org/) - Systems programming language
+- [TigerBeetle](https://tigerbeetle.com/) - Financial accounting database
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [TigerBeetle](https://tigerbeetle.com/) - Financial accounting database written in Zig
```

File: content/podcast/s03e03-oxide/index.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+- [Bun](https://bun.sh/) - JavaScript runtime
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Bun](https://bun.sh/) - JavaScript toolkit written in Zig
```

File: content/podcast/s03e03-oxide/index.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+- [CockroachDB](https://www.cockroachlabs.com/) - Distributed SQL database
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [CockroachDB](https://www.cockroachlabs.com/) - Distributed SQL database used in Oxide's backplane
```

File: content/podcast/s03e03-oxide/index.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+- [Oxide and Friends: Wither CockroachDB?](https://oxide.computer/podcasts/oxide-and-friends/2052742) - Podcast episode
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Oxide and Friends: Wither CockroachDB?](https://oxide.computer/podcasts/oxide-and-friends/2052742) - Discussing Cockroach's switch away from BUSL
```

File: content/podcast/s03e03-oxide/index.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+- [Mozilla Public License](https://choosealicense.com/licenses/mpl-2.0/) - Software license
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Mozilla Public License](https://choosealicense.com/licenses/mpl-2.0/) - Oxide's default software license
```

File: content/podcast/s03e03-oxide/index.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+- [Asahi Linux](https://asahilinux.org/) - Linux on Apple Silicon with Rust drivers
+- [Buck2](https://buck2.build/) - Meta's build system
+- [Jujutsu (jj)](https://martinvonz.github.io/jj/) - Git replacement
+- [Steve's Jujutsu Tutorial](https://steveklabnik.github.io/jujutsu-tutorial/introduction/introduction.html) - Guide to jj
+- [Steve's blog post on branch naming](https://steveklabnik.com/writing/against-names/)
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Steve's blog post on not naming branches](https://steveklabnik.com/writing/against-names/)
```

File: content/podcast/s03e03-oxide/index.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+- [Removing green threads RFC](https://github.com/rust-lang/rfcs/blob/master/text/0230-remove-runtime.md) - Major change in Rust's concurrency model
+- [Hubris](https://github.com/oxidecomputer/hubris) - Oxide's embedded operating system
+- [Tock OS](https://tockos.org/) - Embedded operating system in Rust
+- [cargo-xtask](https://github.com/matklad/cargo-xtask) - Build automation for Rust projects
+- [Hubris Build Documentation](https://github.com/oxidecomputer/hubris#build) - Building Hubris
+- [Buck Build System](https://buck.build/) - Facebook's build system
+- [Buildomat](https://github.com/oxidecomputer/buildomat) - Oxide's build system
+- [Omicron](https://github.com/oxidecomputer/omicron) - Oxide's manufacturing test framework
+- [illumos](https://illumos.org/) - Unix operating system
+- [bhyve](https://bhyve.org/) - Hypervisor
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [bhyve](https://bhyve.org/) - BSD VM hypervisor
```

File: content/podcast/s03e03-oxide/index.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+- [Beowulf cluster](https://en.wikipedia.org/wiki/Beowulf_cluster) - Early parallel computing architecture
+- [Bryan's blog post on Rust](https://bcantrill.dtrace.org/2018/09/18/falling-in-love-with-rust/) - Journey of a systems programmer to Rust
+- [JavaOS](https://en.wikipedia.org/wiki/JavaOS) - Operating system written in Java
+- [D Programming Language](https://dlang.org/) - Systems programming language
+- [Garbage Collection in early Rust](https://pcwalton.github.io/_posts/2013-06-02-removing-garbage-collection-from-the-rust-language.html) - Historical Rust development
+- [Removing green threads RFC](https://github.com/rust-lang/rfcs/blob/master/text/0230-remove-runtime.md) - Major change in Rust's concurrency model
+- [Hubris](https://github.com/oxidecomputer/hubris) - Oxide's embedded operating system
+- [Tock OS](https://tockos.org/) - Embedded operating system in Rust
+- [cargo-xtask](https://github.com/matklad/cargo-xtask) - Build automation for Rust projects
+- [Hubris Build Documentation](https://github.com/oxidecomputer/hubris#build) - Building Hubris
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Hubris Build Documentation](https://github.com/oxidecomputer/hubris#build) - Building Hubris using `cargo xtask`
```

File: content/podcast/s03e03-oxide/index.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+- [I have come to bury the BIOS](https://www.osfc.io/2022/talks/i-have-come-to-bury-the-bios-not-to-open-it-the-need-for-holistic-systems/) - Bryan's talk on firmware
+- [Beowulf cluster](https://en.wikipedia.org/wiki/Beowulf_cluster) - Early parallel computing architecture
+- [Bryan's blog post on Rust](https://bcantrill.dtrace.org/2018/09/18/falling-in-love-with-rust/) - Journey of a systems programmer to Rust
+- [JavaOS](https://en.wikipedia.org/wiki/JavaOS) - Operating system written in Java
+- [D Programming Language](https://dlang.org/) - Systems programming language
+- [Garbage Collection in early Rust](https://pcwalton.github.io/_posts/2013-06-02-removing-garbage-collection-from-the-rust-language.html) - Historical Rust development
+- [Removing green threads RFC](https://github.com/rust-lang/rfcs/blob/master/text/0230-remove-runtime.md) - Major change in Rust's concurrency model
+- [Hubris](https://github.com/oxidecomputer/hubris) - Oxide's embedded operating system
+- [Tock OS](https://tockos.org/) - Embedded operating system in Rust
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [Tock OS](https://tockos.org/) - Embedded operating system in Rust predating Hubris
```

File: content/podcast/s03e03-oxide/index.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+Their 'On The Metal' podcast is a treasure trove of systems programming stories and proudly served as a role model for 'Rust in Production.'
+
+### About Steve Klabnik
+
+In the Rust community, Steve does not need any introduction. He is a prolific writer, speaker, and software engineer who has contributed to the Rust ecosystem in many ways -- including writing the first version of the official Rust book. If you sent a tweet about Rust in the early days, chances are Steve was the one who replied. Previously, he worked at Mozilla and was a member of the Rust and Ruby core teams. 
+
+### Links From The Episode (In Chronological Order)
+
+- [The Rust Programming Language](https://doc.rust-lang.org/book/) ([No Starch Press version](https://nostarch.com/Rust)) - The official Rust book
+- Steve's Rust 1.0 [FOSDEM](https://archive.fosdem.org/2015/schedule/event/the_story_of_rust/) / [ACM](https://www.youtube.com/watch?v=79PSagCD_AY) talk - Early history of Rust
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md]:
```suggestion
- [The Story of Rust - FOSDEM](https://archive.fosdem.org/2015/schedule/event/the_story_of_rust/) / [The History of Rust - ACM](https://www.youtube.com/watch?v=79PSagCD_AY) - Early history of Rust
```

**Review by M3t0r — CHANGES_REQUESTED** [review]

File: content/podcast/s03e03-oxide/index.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+draft = false
+aliases = ["/p/s03e03"]
+[extra]
+guest = "Steve Klabnik"
+role = "Author and Software Engineer"
+season = "03"
+episode = "03"
+series = "Podcast"
++++
+
```
Comment (M3t0r) [inline content/podcast/s03e03-oxide/index.md:14]:
```suggestion

<div><script id="letscast-player-97aa1056" src="https://letscast.fm/podcasts/rust-in-production-82281512/episodes/oxide-s-steve-klabnik/player.js?size=s"></script></div>

```

**Pull request merged**